### DISCUSSION TITLE
Include a "grouped" or Bundled product type

### DISCUSSION THREAD

**Discussion by andris-sevcenko** [craftcms/commerce#2311] [Ideas]
> Created by: Jae Barclay ([jae@ejaedesign.com](jae@ejaedesign.com)) on 2015/05/04 12:25:22 +0000
Votes at time of UserVoice import: 45

------
A highly useful product type (which often is not available in ecomm packages) is the "grouped" product type. 

A grouped product lets you display multiple items of various attribute (say, same flavored candy that comes in boxes of different quantities) in a single page. So you have a single unique URL selling essentially a single product, but it displays buying options for two or more SKU items.

This is in addition to the fairly common product types such as:

1. Single/simple product
2. Configurable product
3. Digital product (downloadable & non-downloadable)
4. Bundled product
5. Extendable product

**Comment by RitterKnightCreative** [comment]
Commenting so this gets more love! 

Bundles/kits are one of the easiest ways to bump a merchant's AOV and it'd be great to see them in Commerce. 

FWIW: here's how [Magento](https://docs.magento.com/m2/ce/user_guide/catalog/product-create-bundle.html) handles it.

We can almost pull off a bundle now using some creativity... eg setup a new "Bundle" product type, relate the products from the bundle into the variant, and do some special Twig magic on the front end. 

Now that Commerce 2 does MultiAdd of purchasables, it's easy to just add all the "product components" of the bundle to the cart. (Could do some additional magic with a "bundle" or "notes" field.)

Depending on what sort of love the discount engine gets in the future, that would be _almost_ enough to tell Commerce "When these 3 products are added to the cart, add them all together and then discount by $X or Y%."

Another problem is how to manage the bundle item's inventory and shipping. If you handle the bundle as one "product" and manage inventory of the bundle's products, you end up setting the bundle price manually as well.  I'm sure a plugin could be made to handle this use case easily enough but a determination would need to be made on how you handle recording the sale... is it a sale of the bundle or a sale of the product... or both? 

It gets sticky fast. 

Internally, Commerce would be in the best position to handle some of the "magic" behind the scenes. when setting up a product type,  I could forsee a special "This is a bundle" option which would automatically close off many of the fields in the editor that aren't needed since the product type is only there as the glue. Then rules inside the product type when setting it up would define how the pricing and inventory are handled (eg options for "remove components from inventory when sold", "determine bundle price by....")

**Comment by bossanova808** [comment]
I'm very keen to see bundle options in Commerce and would love to be involved in the conversation once you start getting in to this.

For us, bundles are primarily groups of products that we tend to sell at a slight discount when bought all together.  At the moment we just create a new product to represent this, but would prefer to be able to do it by choosing products that form the bundle, and setting a bundle price.

I can see the planned condition engine being a big part of this, to represent the bundle pricing rules..for example if 2 of 3 products are included in a sale, then the price is $x, or if 3 of 3, then $y or whatever.

An ideal implementation would be that the bundle can have its own custom fields, but there should also be easy access in the templates to the fields of the constituent products of the bundle of course.

Anyway, great that this is on the radar.
